src/Microsoft.ComponentDetection.Detectors/go/Parsers/GoCLIParser.cs

```diff
@@ -74,19 +74,37 @@ public async Task<bool> ParseAsync(
     private void RecordBuildDependencies(string goListOutput, ISingleFileComponentRecorder singleFileComponentRecorder, string projectRootDirectoryFullName)
     {
         var goBuildModules = new List<GoBuildModule>();
-        var reader = new JsonTextReader(new StringReader(goListOutput))
-        {
-            SupportMultipleContent = true,
-        };
 
         using var record = new GoReplaceTelemetryRecord();
 
-        while (reader.Read())
+        // Go CLI outputs multiple JSON objects (not an array), so we need to parse them one by one
+        var remaining = goListOutput.AsMemory();
+        while (!remaining.IsEmpty)
         {
-            var serializer = new JsonSerializer();
-            var buildModule = serializer.Deserialize<GoBuildModule>(reader);
+            var reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(remaining.ToString()));
+            try
+            {
+                var buildModule = JsonSerializer.Deserialize<GoBuildModule>(ref reader);
+                if (buildModule != null)
+                {
+                    goBuildModules.Add(buildModule);
+                }
+
+                // Move past the consumed bytes plus any whitespace
+                var consumed = (int)reader.BytesConsumed;
+                remaining = remaining[consumed..];
 
-            goBuildModules.Add(buildModule);
+                // Skip whitespace between JSON objects
+                while (!remaining.IsEmpty && char.IsWhiteSpace(remaining.Span[0]))
```

Comment:
Does STJ error if you have starting whitespace? I would have thought that it would silently ignore it.